LanguageFeatures/Patterns/type_inference_A17_t05.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Calculate the static type of the pattern. Using that value type,
+/// recurse through the pattern again downwards to the leaf subpatterns filling
+/// in any missing types in the pattern. This process may also insert implicit
+/// coercions and casts from dynamic when values flow into a pattern during
+/// matching.
+///
+/// @description Check that the calculation of the static type of a list
+/// pattern performs implicit coercions
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "../../Utils/static_type_helper.dart";
+import "../../Utils/expect.dart";
+
+main() {
+  var <double>[x1] = [42];
```

Comment:
Analyzer Ok. Dartk:
```
stderr:
tests/co19/src/LanguageFeatures/Patterns/type_inference_A17_t05.dart:21:15: Error: The matched value of type 'List<int>' isn't assignable to the required type 'List<double>'.
 - 'List' is from 'dart:core'.
Try changing the required type of the pattern, or the matched value type.
  var <double>[x1] = [42];
              ^
```

Reply:
That would be a CFE issue: The type schema from the pattern is `List<double>` and `[42]` inferred with context type `List<double>` should yield `<double>[42.0]`. So keep it.